bqskit/ir/gates/parameterized/rsu3.py

```diff
@@ -0,0 +1,172 @@
+"""This module implements the rotations due to the generatos of SU(3)."""
+from __future__ import annotations
+
+import numpy as np
+import numpy.typing as npt
+
+from bqskit.ir.gates.qutritgate import QutritGate
+from bqskit.qis.unitary.differentiable import DifferentiableUnitary
+from bqskit.qis.unitary.unitary import RealVector
+from bqskit.qis.unitary.unitarymatrix import UnitaryMatrix
+from bqskit.utils.cachedclass import CachedClass
+from bqskit.utils.math import SUGen
+from bqskit.utils.typing import is_integer
+
+
+class RSU3Gate(QutritGate, DifferentiableUnitary, CachedClass):
+    """Rotation by SU3 generator for a single qutrit gate."""
```

Comment:
This is a super cool gate! I know you told me you were interested in generalizing this to SU(N), and I am very happy to discuss this further.

I know there are infinitely many ways to generate SU(3), are these the gell-mann matrices? I think it would interesting to explore the different options and the related effects on encoding on synthesis.

I am also curious if we can expand this to create combinations/multi-qudit versions of them. Similar to how you might want an XX+YY gate.